WIP: Tab Redesign

jannuary opened the pull request:
Ephy uses HdyTabBar now, it would be nice to have styling that mimics it for consistency here.

Pretty dirty and unfinished, and only works with the light theme now.

- [x] Clean things up
- [ ] Dark, HC, Yaru, etc. colors
- [ ] Auto hide close button (properly)
- [x] Lock behind a preference
![image](https://user-images.githubusercontent.com/27908024/107940382-b79c3500-6fba-11eb-8a27-2ef80633b28b.png)

rafaelmardojai commented:
Looks good! 

We should put these changes behind a preference like `gnomeTheme.newTabs` before making them default.

jannuary commented:
Hm, I might need help implementing the close button autohide properly, here's how Web does it:
![Peek 2021-02-22 19-28](https://user-images.githubusercontent.com/27908024/108708693-5e954980-7544-11eb-9d8e-458b09db4a02.gif)

jannuary commented:
Can't particularly test alternative color schemes or implement close-button autohide at the moment, so unmarking as WIP and requesting a review.

rafaelmardojai commented on theme/parts/tabsbar.css:
```diff
@@ -316,3 +316,70 @@ tab[selected]:-moz-window-inactive {
+		background-size: 24px;
+		background-color: var(--gnome-tabbar-close-hover);
+		box-shadow: 0px 0px transparent; 
+	    border: 0px solid transparent;
```
This should be applied to `.tab-close-button` it self or will do a jump on hover.

rafaelmardojai commented:
Is this what you want?
![output](https://user-images.githubusercontent.com/6210397/114476441-d20c1c80-9bbf-11eb-987a-df94ac099965.gif)

kdwk commented:
Can't wait to try this out

jannuary commented:
Marking as a draft because Firefox updated and broke this. Apologies for letting this PR sit around for months.

rafaelmardojai commented:
@jannuary I'm thinking on integrating this directly to the `beta` branch. I have some drafts locally. But no worry, I will give you the credit on the commit :P

jannuary commented:
Rebased on the beta branch, fixed couple of things, close button autohide still buggy though:
![image](https://user-images.githubusercontent.com/27908024/117607544-613c2e00-b186-11eb-8930-2f86ffc748c6.png)

rafaelmardojai commented:
Nice, I think we can remove the preference and apply the change directly, with those Firefox breakages we don't want to maintain two tabs styles 😅.

Could you also rebase again and merge against `beta`?
